crates/astria-core/src/protocol/bridge/v1alpha1/mod.rs

```diff
@@ -76,3 +83,167 @@ enum BridgeAccountLastTxHashResponseErrorKind {
+    /// - if the `rollup_id` field is set but the `withdrawer_address` field is not
+    /// - if the `rollup_id` field is set but the `asset_id` field is not
+    /// - if the `asset_id` field does not contain a valid asset ID
+    /// - if the `rollup_id` field is set but invalid
+    /// - if the `sudo_address` field is set but invalid
+    /// - if the `withdrawer_address` field is set but invalid
+    pub fn try_from_raw(
+        raw: raw::BridgeAccountInfoResponse,
+    ) -> Result<Self, BridgeAccountInfoResponseError> {
+        let raw::BridgeAccountInfoResponse {
+            height,
+            rollup_id,
+            asset_id,
+            sudo_address,
+            withdrawer_address,
+        } = raw;
+
+        let Some(rollup_id) = rollup_id else {
+            return Ok(Self {
+                height,
+                info: None,
+            });
+        };
+
+        let Some(sudo_address) = sudo_address else {
+            return Err(BridgeAccountInfoResponseError::field_not_set(
+                "sudo_address",
+            ));
+        };
+
+        let Some(withdrawer_address) = withdrawer_address else {
+            return Err(BridgeAccountInfoResponseError::field_not_set(
+                "withdrawer_address",
+            ));
+        };
+
+        let Some(asset_id) = asset_id else {
+            return Err(BridgeAccountInfoResponseError::field_not_set("asset_id"));
+        };
+
+        let asset_id = asset::Id::try_from_slice(&asset_id)
+            .map_err(BridgeAccountInfoResponseError::asset_id)?;
+
+        Ok(Self {
+            height,
+            info: Some(BridgeAccountInfo {
+                rollup_id: RollupId::try_from_raw(&rollup_id)
+                    .map_err(BridgeAccountInfoResponseError::invalid_rollup_id)?,
+                asset_id,
+                sudo_address: Address::try_from_raw(&sudo_address)
+                    .map_err(BridgeAccountInfoResponseError::invalid_sudo_address)?,
+                withdrawer_address: Address::try_from_raw(&withdrawer_address)
+                    .map_err(BridgeAccountInfoResponseError::invalid_withdrawer_address)?,
+            }),
+        })
+    }
+
+    #[must_use]
+    pub fn into_raw(self) -> raw::BridgeAccountInfoResponse {
+        let Some(info) = self.info else {
+            return raw::BridgeAccountInfoResponse {
+                height: self.height,
+                rollup_id: None,
+                asset_id: None,
+                sudo_address: None,
+                withdrawer_address: None,
+            };
+        };
+
+        raw::BridgeAccountInfoResponse {
+            height: self.height,
+            rollup_id: Some(info.rollup_id.into_raw()),
+            asset_id: Some(info.asset_id.get().to_vec()),
+            sudo_address: Some(info.sudo_address.into_raw()),
+            withdrawer_address: Some(info.withdrawer_address.into_raw()),
+        }
+    }
+}
+
+impl raw::BridgeAccountInfoResponse {
```

Comment:
It was pretty annoying having to update all these `.into_native`/`from_native` methods. I think I introduced these utilities prematurely and we should get rid of them/not introduce them for now (also since thehy are only used in `astria-sequencer-client`).